src/wallet/rpcwallet.cpp

```diff
@@ -2598,9 +2599,20 @@ static UniValue createwallet(const JSONRPCRequest& request)
         flags |= WALLET_FLAG_DISABLE_PRIVATE_KEYS;
     }
 
+    bool create_blank = false; // Indicate that we the wallet is actually supposed to be blank
     if (!request.params[2].isNull() && request.params[2].get_bool()) {
+        create_blank = true;
         flags |= WALLET_FLAG_BLANK_WALLET;
     }
+    SecureString passphrase;
+    passphrase.reserve(100);
+    if (!request.params[3].isNull()) {
+        passphrase = request.params[3].get_str().c_str();
+        if (!passphrase.empty()) {
```

Comment:
I think it would be safer raise an error if the passphrase is non-null and empty instead of pretending the passphrase wasn't provided. I'm imagining a buggy shell script with `bitcoin-cli createwallet "$NAME" false false "$PASS"` that forgot to set the PASS variable or spelled it differently. Better to fail in this case than go on like nothing's wrong.

Reply:
Done